waftools/detections/compiler_swift.py

```diff
@@ -6,7 +6,9 @@ def __run(cmd):
                                      stdout=Utils.subprocess.PIPE,
                                      stderr=Utils.subprocess.PIPE,
                                      shell=True)
-        output = cmd.stdout.read().decode().strip()
+        output = cmd.stdout.read().strip()
+        if type(output) is not str:
+            output = output.decode()
```

Comment:
alternatively, passing `universal_newlines=True` to Popen() will force streams to be opened in text mode

also you're opening a pipe for stderr but never reading from it

Reply:
indeed. thx this seems a lot better and cleaner.